gwnr/waveform/hybridize.py

```diff
@@ -0,0 +1,177 @@
+        where the frq values are within the tolerance wrt desired frq, 
+        there will be multiple such values '''
+
+        where_i1 = np.where(i1_tentative)
+        ''' where_i1 is a tuple that holds a small array of the index values of frq_timeseries,
+        corresponding to the indices where i1_tentative is TRUE '''
+
+        cost_i1 = np.square(frq_timeseries[where_i1] - frq_desired)
+        ''' cost_i1 is a cost fn which minimised the square of the difference between the frq values 
+        at the indices provided by where_i1, and the desired frq, this will have the length same as the array held by the tuple where_i1 '''
+
+        i1_forsure = np.where(cost_i1 == np.min(cost_i1))
+        ''' i_forsure is a tuple that holds an array which holds the numerical value of the index where cost_i1 is minimum '''
+
+        i1 = where_i1[0][i1_forsure[0][0]] 
+        ''' i1 is an integer, which is the numerical value of the index value where frq_desired is located, 
+        it is extracted from the tuple where_i1'''
+        return i1
+    
+    
+    
+    t1_index_insp = find_i1(frq_insp, frq_attach - frq_width/2)
+    t2_index_insp = find_i1(frq_insp, frq_attach + frq_width/2)
+
+
+    t1_index_mr_tent = find_i1(frq_mr, frq_attach - frq_width/2)
+    # t2_index_mr_tent = find_i1(frq_mr, frq_attach + frq_width/2) 
+
+    # another way to define t2_index_mr_tent is through number of points in the inspiral window    
+    t2_index_mr_tent = t1_index_mr_tent + (t2_index_insp-t1_index_insp)
+
+    no_cycles = (phase_insp[t1_index_insp] - phase_insp[t2_index_insp])/(2*np.pi)
+
+
+    ''' After alignment, need to find corresponding indices for merger_ringdown as well '''
+    
+    
+    ''' Alignment: '''
+    
+    # Defining the Comb
+    
+    def get_comb(t1_index_insp, t2_index_insp, no_sp):
+
+        d = int((t2_index_insp-t1_index_insp)/no_sp)
+
+        count1 = np.empty(no_sp-1)
+        for kk in range(no_sp-1):
+            count1[kk] = kk*d
+
+        samples = np.append(count1, t2_index_insp-t1_index_insp-1).astype(int)
+        return samples
+
+    sample_indices_insp = get_comb(t1_index_insp, t2_index_insp, no_sp = 4)
+    sample_indices_mr = get_comb(t1_index_mr_tent, t2_index_mr_tent, no_sp = 4)
+
+
+
+    ''' Enter alignment code here '''
+
+    ''' Need interpolated frequency evaluated on the time axis for timeshifts '''
+
+    interp_phase_insp = UnivariateSpline(timeaxis_insp[t1_index_insp:t2_index_insp], phase_insp[t1_index_insp:t2_index_insp], k = 4)
+    interp_phase_mr = UnivariateSpline(timeaxis_mr[t1_index_mr_tent:t2_index_mr_tent], phase_mr[t1_index_mr_tent:t2_index_mr_tent], k = 4)
+
+    interp_frq_insp = interp_phase_insp.derivative()(timeaxis_insp[t1_index_insp:t2_index_insp])/(2*np.pi)
+    interp_frq_mr = interp_phase_mr.derivative()(timeaxis_mr[t1_index_mr_tent:t2_index_mr_tent])/(2*np.pi)
+
+
+
+
+
+
+
+
+
+    ''' alignment using corrective phase addition '''
+
+    def mismatch_discrete(w1, w2):
```

Comment:
Move this function out.

Reply:
rename to something like `l2_norm_difference_of_two_series`

Reply:
Calling it L2 norm will be misleading as we are not taking the sqrt of the quantity (squared difference), and the normalisation is through the first quantity.